src/averageAges.js

```diff
@@ -56,6 +78,22 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  let filteredKids = people.filter(kid => people.some(person =>
+    kid.mother === person.name));
+
+  filteredKids = onlyWithSon
+    ? filteredKids.filter(kid => kid.sex === 'm')
+    : filteredKids;
+
```

Comment:
it's better to check onlyWithSon inside the first filter and make filteredKids `const`